sphinx_needs/lsp/esbonio.py

```diff
@@ -11,15 +11,29 @@
 from esbonio.lsp.rst import CompletionContext, DefinitionContext, HoverContext
 from esbonio.lsp.sphinx import SphinxLanguageServer
 from jinja2 import BaseLoader, Environment
-from pygls.lsp.types import (
-    CompletionItem,
-    CompletionItemKind,
-    InsertTextFormat,
-    Location,
-    Position,
-    Range,
-    TextEdit,
-)
+from pygls import __version__
+
+if __version__ < "1.0":
```

Comment:
I think this will not work.
As "1.0" is a string and not something numeric, where you can use <.

Maybe the above solution from here works better:
https://stackoverflow.com/a/11887885/1096526

Reply:
It works. pygls \__version__ is a string, not numeric.  https://github.com/openlawlibrary/pygls/blob/master/pygls/__init__.py

Reply:
Yes, but the string comparison checks the first character only. 
There are a lot of cases where this does not work and produces wrong results:
![image](https://user-images.githubusercontent.com/998700/217476824-f965788b-1fc9-4740-a13d-f5a979c2dd80.png)

To be save, we need to parse the version-string and get a safe, comparable version-object out of it.